pkg/networkservice/mechanisms/memif/common.go

```diff
@@ -33,60 +37,96 @@ import (
 	"github.com/networkservicemesh/api/pkg/api/networkservice/payload"
 	"github.com/networkservicemesh/sdk/pkg/tools/log"
 	"github.com/pkg/errors"
+	"github.com/vishvananda/netns"
+	"golang.org/x/sys/unix"
 
 	"github.com/networkservicemesh/sdk-vpp/pkg/networkservice/up"
 	"github.com/networkservicemesh/sdk-vpp/pkg/tools/ifindex"
 )
 
-var lastSocketID uint32
+type vppConnection struct {
+	isExternal bool
+
+	api.Connection
+}
+
+var (
+	netNS     netns.NsHandle
+	netNSPath string
+	once      sync.Once
+)
```

Comment:
@Mixaster995 these should not be global variables, they should be attributes of the chain(s) that use them.

Reply:
@edwarnicke made refactoring to remove usage of global variables - please, take a look